### ISSUE TITLE
Tiny loss during training, but no detections, even on training set

### ISSUE THREAD

**Issue by cesare-montresor** [pjreddie/darknet#795]
I'm trying to train a detector on my own dataset, currently made of just one class, drones.

- I converted an existing dataset into VOC annotations, parsed it using `voc_label.py`.
- Set the voc.names to just one line, "drone"
- changed the `yolov3.cfg`: `batch=32, subdivision=8, classes=1 and filters=18`
- run `./darket detector train cfg/voc.names cfg/yolov3.cfg darknet53.conv.74`

The training converges and the loss from more than 1000 decrease to 0.x
When I run the detector I have always 0 detections, if I try with `-thresh 0` many box appears, all of them at 0% ( in terminal ) 

Any suggestion ?

**Comment by cesare-montresor** [comment]
EDIT: Update, using the very same settings I changed dataset and it actually trains, does it mean that the dataset I'm trying to use is rubbish? Perhaps I'm noticing that is black and white and quite noisy ( video take from a flying drone )

**Comment by hongrui16** [comment]
I have  the same question as you. I run the training on the VOC datasets , then i use the generated weight file for the test.
I run the command ''./darknet detector test cfg/voc.data cfg/yolov3-voc.cfg yolov3-voc.weights data/dog.jpg'', but there is no object been detected. It is not a matter of threshold because i change its value to 0.
I am trying find out the solution for this problem. If there is any progress, please let us know.

**Comment by cesare-montresor** [comment]
UPDATE: So far, after changing dataset, detections started to improve, so I guess the problem was just a very poor dataset.

**Comment by hongrui16** [comment]
Good news. 
At first I thought it might be some bugs in the training code as the VOC datasets is frequently being used. I review the code, but got a conclusion in a short time. So  I decided to train the YOLO3 with VOC in another version ''https://github.com/experiencor/keras-yolo3''. There has been a weight file generated and I have used it for test. There are some objects been detected, though the results are not precise.

The training label is generated in this commander "cat 2007_train.txt 2007_val.txt 2012_*.txt > train.txt
"
So I wonder the annotations is not been taken in the training progress in Darknet. That is there is no bounding box, so there are no correct weight parameters. While, this wonder need to be proved.

**Comment by hainan89** [comment]
> Good news.
> At first I thought it might be some bugs in the training code as the VOC datasets is frequently being used. I review the code, but got a conclusion in a short time. So I decided to train the YOLO3 with VOC in another version ''https://github.com/experiencor/keras-yolo3''. There has been a weight file generated and I have used it for test. There are some objects been detected, though the results are not precise.
> 
> The training label is generated in this commander "cat 2007_train.txt 2007_val.txt 2012_*.txt > train.txt
> "
> So I wonder the annotations is not been taken in the training progress in Darknet. That is there is no bounding box, so there are no correct weight parameters. While, this wonder need to be proved.

Hi @hongrui16 

I got the same issue, did you get any new idea?
I  also notice that the label data seems not be feeded to the traing process。
At first, I just suppose that the label file should be located in the same folder with the image folder.
However, when I train the VOC dataset following the official instructions, the label fiels do not located in the same folder with the images, but the traing process can still be conducted withou alert?

I think this is the point? 
Hope you have already fixed this issue, do you have any comments? Thanks.

**Comment by hongrui16** [comment]
@hainan89  sorry, I almost forget this problem and i have not had any ideas yet. If i get some idea, I will tell you.

**Comment by ludwikbukowski** [comment]
the same problem
